internal/resources/google/container_cluster.go

```diff
@@ -76,8 +74,7 @@ func (r *ContainerCluster) BuildResource() *schema.Resource {
 		subresources = append(subresources, r.DefaultNodePool.BuildResource())
 	}
 
-	for i, nodePool := range r.NodePools {
-		nodePool.Address = fmt.Sprintf("node_pool[%d]", i)
```

Comment:
On line 75 I'm setting the address for the default pool too:
```go
r.DefaultNodePool.Address = "default_pool"
```

Guess we can remove it as well, right? With `apply` it will be "default-pool".